quetz/dao.py

```diff
@@ -409,16 +409,28 @@ def create_package_member(self, channel_name, package_name, new_member):
         self.db.commit()
 
     def get_api_keys_with_members(self, user_id):
-        return (
+        user_role_api_keys = (
+            self.db.query(ApiKey)
+            .select_from(ApiKey)
+            .filter(ApiKey.owner_id == user_id)
+            .filter(ApiKey.user_id == user_id)
+            .filter(~ApiKey.deleted)
+            .all()
+        )
+
+        custom_role_api_keys = (
             self.db.query(ApiKey, PackageMember, ChannelMember)
             .select_from(ApiKey)
             .filter(ApiKey.owner_id == user_id)
             .filter(~ApiKey.deleted)
+            .filter(ApiKey.owner_id != ApiKey.user_id)
```

Comment:
probably that's not enough, because a user could create an api key for a different user. Altough at the moment you can do that with web API.

you could instead check whether a profile is not null with something like this (probably it would need some tweaking):

```
...
.outerjoin(Profile, ApiKey.user_id == Profile.user_id).filter(Profile.user_id != None)
...
```